File: Algorithms/Java/Palindrome Number.java
public class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
			return false;
		}
		long origin = (long) x;
		long res = 0;
		while (origin != 0) {
			res = 10 * res + origin % 10;
			origin /= 10;
		}
		return res == x;
    }
}

File: Algorithms/Python/Palindrome-Number.py
class Solution(object):

    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0:
            return False
        ans, origin = 0, x
        while x != 0:
            ans = 10 * ans + x % 10
            x //= 10
        return ans == origin
